src/ifcopenshell-python/ifcopenshell/ids.py

```diff
@@ -1276,52 +1165,42 @@ def emit(self, log_content):
         :param log_content: default logger message
         :type log_content: string|dict
         """
+        data = log_content.msg
+        isValid = isinstance(data, dict)
+        if not isValid:
+            return
         topic = bcf.Topic()
-        topic.title = log_content.msg["sentence"].split(".\n")[1]
-        topic.description = log_content.msg["sentence"].split(".\n")[0]
+        topic.title = data["title"]
+        topic.description = data["description"]
         self.bcf.add_topic(topic)
-        # try:  # Add viewpoint and link to ifc object
         viewpoint = bcf.Viewpoint()
         viewpoint.perspective_camera = bcf.PerspectiveCamera()
-        ifc_elem = log_content.msg["ifc_element"]
-        # ifc_elem = ifc_file.by_guid(log_content.msg["guid"])
+
+        ifc_elem = data["ifc_element"]
         target_position = np.array(ifcopenshell.util.placement.get_local_placement(ifc_elem.ObjectPlacement))
         target_position = target_position[:, 3][0:3]
-        camera_position = target_position + np.array((5, 5, 5))
+        target_position /= 1000 # convert to meters
```

Comment:
This hardcoded unit conversion doesn't quite look right. Perhaps use something in ifcopenshell.util.unit to help convert?